owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewFormatTextFragmentStateAdapter.kt

```diff
@@ -0,0 +1,103 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Parneet Singh
+ *
+ * Copyright (C) 2023 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.ui.preview
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import com.owncloud.android.R
+import io.noties.markwon.Markwon
+import io.noties.markwon.ext.strikethrough.StrikethroughPlugin
+import io.noties.markwon.ext.tables.TablePlugin
+import io.noties.markwon.ext.tasklist.TaskListPlugin
+import io.noties.markwon.html.HtmlPlugin
+
+class PreviewFormatTextFragmentStateAdapter(
+    fragment: Fragment, private val text: String, private val mimeType: String
```

Comment:
Hi, yes it's not being used anywhere outside. But it isn't being exposed either, because it's not a property and we can access value parameters (as [kotlin docs](https://kotlinlang.org/docs/classes.html#constructors:~:text=Primary%20constructor%20parameters%20can%20be%20used%20in%20the%20initializer%20blocks.%20They%20can%20also%20be%20used%20in%20property%20initializers%20declared%20in%20the%20class%20body%3A) says) of a primary constructor inside init block as well as class level property initializer (in this case it would be:
```
val formatTypes =
        mapOf(TYPE_MARKDOWN to fragment.getString(R.string.tab_label_markdown), TYPE_PLAIN to fragment.getString(R.string.tab_label_ascii))
```
So, It's just like a local parameter of constructor (like in java). Hence I cannot mark it as private and if I want to it needs to be a property which in fact it's not being used as.
- If I set it private only, IDE (error) -> `Constructor parameter is never used as a property` 
- and if i set it `private val ..`, IDE (warning with `val` grayed out) -> Constructor parameter is never used as a property

Reply:
Then we'll follow Kotlin rules! 😀